src/uu/ls/src/ls.rs

```diff
@@ -625,20 +660,31 @@ impl Config {
 }
 
 #[uucore_procs::gen_uumain]
+#[allow(unused_mut)]
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     let usage = usage();
 
     let app = uu_app().usage(&usage[..]);
 
     let matches = app.get_matches_from(args);
 
-    let config = Config::from(&matches)?;
+    let mut config = Config::from(&matches)?;
+
+    #[cfg(unix)]
+    {
+        // There may be more formatting issues like this.
+        // Be on the look out!
+        if config.inode && config.format == Format::OneLine {
+            config.format = Format::Columns;
+        }
+    }
+
     let locs = matches
         .values_of_os(options::PATHS)
         .map(|v| v.map(Path::new).collect())
         .unwrap_or_else(|| vec![Path::new(".")]);
 
-    list(locs, config)
+    list(locs, &config)
```

Comment:
This change is fine, but I think it might be based on a misconception. Borrowing in Rust is not necessarily faster than moving. In other languages, the `&` might refer to getting a pointer, but in Rust that's not really the case, it is really only about ownership. This means that in this case, the compiler will probably just pass by reference anyway even if you "move" the value (i.e. give ownership of the config to the `list` function)

Reply:
I didn't know all of that, and I appreciate you letting me know.